packages/core/src/TxBuilders/TxBuilder.Lucid.V3.class.ts

```diff
@@ -324,42 +316,119 @@ export class TxBuilderLucidV3 extends TxBuilder {
       );
     }
 
+    const sortedAssets = SundaeUtils.sortSwapAssetsWithAmounts([
+      assetA,
+      assetB,
+    ]);
+
+    const exoticPair = !SundaeUtils.isAdaAsset(sortedAssets[0].metadata);
+
     const [userUtxos, { hash: poolPolicyId }, references, settings] =
       await Promise.all([
-        this.getUtxosForPoolMint([assetA, assetB]),
+        this.getUtxosForPoolMint(),
         this.getValidatorScript("pool.mint"),
         this.getAllReferenceUtxos(),
         this.getAllSettingsUtxos(),
       ]);
 
+    const newPoolIdent = DatumBuilderLucidV3.computePoolId(userUtxos[0]);
+
+    const poolNftNameHex = toUnit(
+      poolPolicyId,
+      DatumBuilderLucidV3.computePoolNftName(newPoolIdent)
+    );
+    const poolRefNameHex = toUnit(
+      poolPolicyId,
+      DatumBuilderLucidV3.computePoolRefName(newPoolIdent)
+    );
+    const poolLqNameHex = toUnit(
+      poolPolicyId,
+      DatumBuilderLucidV3.computePoolLqName(newPoolIdent)
+    );
+
+    const poolDepositAddress = await this.generateScriptAddress("pool.mint");
+
+    const assetsDeposited = C.MultiAsset.new();
+    assetsDeposited.set_asset(
+      C.ScriptHash.from_hex(poolPolicyId),
+      C.AssetName.new(
+        Buffer.from(DatumBuilderLucidV3.computePoolNftName(newPoolIdent), "hex")
+      ),
+      C.BigNum.from_str("1")
+    );
+    assetsDeposited.set_asset(
+      C.ScriptHash.from_hex(sortedAssets[1].metadata.assetId.split(".")[0]),
+      C.AssetName.new(
+        Buffer.from(sortedAssets[1].metadata.assetId.split(".")[1])
+      ),
+      C.BigNum.from_str(sortedAssets[1].amount.toString())
+    );
+    if (exoticPair) {
+      assetsDeposited.set_asset(
+        C.ScriptHash.from_hex(sortedAssets[0].metadata.assetId.split(".")[0]),
+        C.AssetName.new(
+          Buffer.from(sortedAssets[0].metadata.assetId.split(".")[1])
+        ),
+        C.BigNum.from_str(sortedAssets[0].amount.toString())
+      );
+    }
+
+    const minDepositLovelace = BigInt(
+      C.min_ada_required(
+        C.TransactionOutput.new(
+          C.Address.from_bech32(poolDepositAddress),
+          C.Value.new_from_assets(assetsDeposited)
+        ),
+        C.BigNum.from_str(
+          (
+            await this.lucid.provider.getProtocolParameters()
+          ).coinsPerUtxoByte.toString()
+        )
+      ).to_str()
+    );
+
+    const depositFee = exoticPair ? minDepositLovelace : 0n;
```

Comment:
I think the logic you want here is:
```suggestion
    const depositFee = exoticPair ? minDepositLovelace : Math.max(minDepositLovelace - sortedAssets[0].amount, 0);
```